src/main/python/pre_fx_data.py

```diff
@@ -0,0 +1,269 @@
+
+    qc_criteria_set = pyqtSignal(dict, name="qc_criteria_set")
+    qc_criteria_unset = pyqtSignal(name="qc_criteria_unset")
+
+    begin_commit_calculated = pyqtSignal(name="begin_commit_calculated")
+    end_commit_calculated = pyqtSignal(name="end_commit_calculated")
+
+    def __init__(self):
+        """ Main data store for all data upstream of feature extraction. This
+        includes:
+            - the EphysDataSet
+            - the StimulusOntology
+            - the qc criteria
+            - the sweep extraction results
+            - the qc results
+        """
+        super(PreFxData, self).__init__()
+
+        self._stimulus_ontology: Optional[StimulusOntology] = None
+        self._qc_criteria: Optional[Dict] = None
+        self.data_set: Optional[EphysDataSet] = None
+        self.nwb_path: Optional[str] = None
+
+    def _notifying_setter(
+        self, 
+        attr_name: str, 
+        value: Any, 
+        on_set: pyqtSignal, 
+        on_unset: pyqtSignal,
+        send_value: bool = False
+    ):
+        """ Utility for a setter that emits Qt signals when the attribute in 
+        question changes state.
+
+        Parameters
+        ----------
+        attr_name :
+            identifies attribute to be set
+        value : 
+            set attribute to this value
+        on_set : 
+            emitted when the new value is not None
+        on_unset :
+            emitted when the new value is None
+        send_value : 
+            if True, the new value will be included in the emitted signal
+
+        """
+        setattr(self, attr_name, value)
+
+        if value is None:
+            on_unset.emit()
+        else:
+            if send_value:
+                on_set.emit(value)
+            else:
+                on_set.emit()
+
+    @property
+    def stimulus_ontology(self) -> Optional[StimulusOntology]:
+        return self._stimulus_ontology
+
+    @stimulus_ontology.setter
+    def stimulus_ontology(self, value: Optional[StimulusOntology]):
+        self._notifying_setter(
+            "_stimulus_ontology", 
+            value,
+            self.stimulus_ontology_set, 
+            self.stimulus_ontology_unset,
+            send_value=True
+        )
+
+    @property
+    def qc_criteria(self) -> Optional[Dict]:
+        return self._qc_criteria
+
+    @qc_criteria.setter
+    def qc_criteria(self, value: Optional[Dict]):
+        self._notifying_setter(
+            "_qc_criteria", 
+            value,
+            self.qc_criteria_set, 
+            self.qc_criteria_unset,
+            send_value=True
+        )
+
+    def set_default_stimulus_ontology(self):
+        self.load_stimulus_ontology_from_json(
+            StimulusOntology.DEFAULT_STIMULUS_ONTOLOGY_FILE
+        )
+
+    def set_default_qc_criteria(self):
+        self.load_qc_criteria_from_json(DEFAULT_QC_CRITERIA_FILE)
+
+    def load_stimulus_ontology_from_json(self, path: str):
```

Comment:
It seems odd that we are starting processing of data in the function called `load_stimulus_ontology_from_json`. It would  be clearer it this function did just handle the ontology.

Reply:
I think you're right about separation of concerns. However:
- we want loading a new stimulus ontology to kick off new sweep extraction and auto qc. Otherwise you could end up in a mismatched state
-  we want loading a new ontology to fail if the ontology is broken in some way. We can only assess this by running processing.

The result of the current implementation is that if the user loads a new stimulus ontology and something then goes wrong in processing, the state is reverted and a message is emitted that explicitly notes the new stimulus ontology as the origin of the issue.

Reply:
per in-person conversation: we are going with present behavior